rir/src/ir/CodeStream.h

```diff
@@ -230,6 +230,80 @@ class CodeStream {
         sources.back() = idx;
     }
 
+#ifdef GNUR_COMPATIBILITY
+
+    unsigned currentPos() const {
+        return pos;
+    }
+
+    unsigned currentSourcesSize() const {
+        return sources.size();
+    }
+
+    void insertIntoStream(BC instr, unsigned beforePos, unsigned sourcesIdx) {
```

Comment:
I find it not a super good idea to insert into the code stream. There is another class called a code-editor that has this functionality. Also it seems a bit of a questionable design decision to me. I would have assumed that you do one of the two:
1. Find out beforehand by analyzing the AST and create code accordingly.
2. Find out after the fact by analyzing the bytecode and removing the context if unnecessary (this is what you could do in an analysis pass).

I guess 1 is hard to do, because it depends on the optimizations that the compiler does. and 2 is annoying because of how the for loop juggles the context on the stack.

Would you be interested in reimplementing it in an optimization pass? I could help you by removing the ugly stack juggling code from the for loop (I wanted to do this for long). Then the only change needed would be to remove the create-context and end-context instructions.

If you find that overblown I would suggest to have an override instruction function on the stream and replace the context creation by 'nop' instructions (that we'd still need to add). At least I would really prefer not to add the ability to insert into the stream.